### PR TITLE
Tidy up logging

### PR DESCRIPTION
**Pull request by willcl-ark**
* Tidy up the overly-verbose tank logging.

* Modernise config with `dictConfig` style logging from json config file:
  * Send DEBUG <= n < WARNING to `stdout`
  * Send >= WARNING to `stderr`
  * Write >= DEBUG to `log_file`, with extra info

### CODE REVIEW

File: src/warnet/tank.py [hunk 1]
```diff
@@ -59,12 +59,13 @@ def _parse_version(self):
 
     def parse_graph_node(self, node):
         # Dynamically parse properties based on the schema
+        graph_properties = {}
         for property, specs in self.warnet.node_schema["properties"].items():
             value = node.get(property, specs.get("default"))
             if property == "version":
                 self._parse_version()
             setattr(self, property, value)
-            logger.debug(f"{property}={value}")
```
Comment (pinheadmz) [inline src/warnet/tank.py:70]: phew thanks!

Comment (willcl-ark) [reply src/warnet/tank.py:70]: Right!

File: src/warnet/utils.py [hunk 2]
```diff
@@ -35,6 +35,12 @@
 NODE_SCHEMA_PATH = TEMPLATES / "node_schema.json"
 
 
+class NonErrorFilter(logging.Filter):
```
Comment (pinheadmz) [inline src/warnet/utils.py:39]: So, just to be clear this means the log will ONLY record errors?

```json
  "filters": {
    "no_errors": {
      "()": "warnet.utils.NonErrorFilter"
    }
```

Comment (willcl-ark) [reply src/warnet/utils.py:39]: It will record NOT errors :)

So everything < WARNING (aka <= INFO)

Comment (willcl-ark) [reply src/warnet/utils.py:39]: Used to send warnings (and above) to stderr, and < warnings to stdout. Not strictly essential for us.

**Comment by pinheadmz** [comment]
Seeing this big response blobs when running  tests in K8s, cant figure out why. They look like urllib3 logs but i can see youve set that module to CRITICAL in the config



> 2024-03-04T15:20:30+0000 - DEBUG - response body: {"kind":"Pod","apiVersion":"v1","metadata":{"name":"warnet-test-gwdzepwv-tank-000003","namespace":"warnet","uid":"39b7b73d-3e8e-4041-bcbe-0f7bf1dd85ae","resourceVersion":"1472746","creationTimestamp":"2024-03-04T15:20:20Z","labels":{"app":"warnet-test-gwdzepwv-tank-000003","network":"warnet-test-gwdzepwv"},"managedFields":[{"manager":"OpenAPI-Generator","operation":"Update","apiVersion":"v1","time":"2024-03-04T15:20:20Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:labels":{".":{},"f:app":{},"f:network":{}}},"f:spec":{"f:containers":{"k:{\"name\":\"bitcoin\"}":{".":{},"f:env":{".":{},"k:{\"name\":\"BITCOIN_ARGS\"}":{".":{},"f:name":{},"f:value":{}}},"f:image":{},"f:imagePullPolicy":{},"f:livenessProbe":{".":{},"f:exec":{".":{},"f:command":{}},"f:failureThreshold":{},"f:initialDelaySeconds":{},"f:periodSeconds":{},"f:successThreshold":{},"f:timeoutSeconds":{}},"f:name":{},"f:readinessProbe":{".":{},"f:failureThreshold":{},"f:periodSeconds":{},"f:successThreshold":{},"f:tcpSocket":{".":{},"f:port":{}},"f:timeoutSeconds":{}},"f:resources":{},"f:securityContext":{".":{},"f:capabilities":{".":{},"f:add":{}},"f:privileged":{}},"f:terminationMessagePath":{},"f:terminationMessagePolicy":{}}},"f:dnsPolicy":{},"f:enableServiceLinks":{},"f:restartPolicy":{},"f:schedulerName":{},"f:securityContext":{},"f:terminationGracePeriodSeconds":{}}}},{"manager":"kubelet","operation":"Update","apiVersion":"v1","time":"2024-03-04T15:20:21Z","fieldsType":"FieldsV1","fieldsV1":{"f:status":{"f:conditions":{"k:{\"type\":\"ContainersReady\"}":{".":{},"f:lastProbeTime":{},"f:lastTransitionTime":{},"f:status":{},"f:type":{}},"k:{\"type\":\"Initialized\"}":{".":{},"f:lastProbeTime":{},"f:lastTransitionTime":{},"f:status":{},"f:type":{}},"k:{\"type\":\"Ready\"}":{".":{},"f:lastProbeTime":{},"f:lastTransitionTime":{},"f:status":{},"f:type":{}}},"f:containerStatuses":{},"f:hostIP":{},"f:phase":{},"f:podIP":{},"f:podIPs":{".":{},"k:{\"ip\":\"10.1.3.54\"}":{".":{},"f:ip":{}}},"f:startTime":{}}},"subresource":"status"}]},"spec":{"volumes":[{"name":"kube-api-access-7fmrb","projected":{"sources":[{"serviceAccountToken":{"expirationSeconds":3607,"path":"token"}},{"configMap":{"name":"kube-root-ca.crt","items":[{"key":"ca.crt","path":"ca.crt"}]}},{"downwardAPI":{"items":[{"path":"namespace","fieldRef":{"apiVersion":"v1","fieldPath":"metadata.namespace"}}]}}],"defaultMode":420}}],"containers":[{"name":"bitcoin","image":"bitcoindevproject/bitcoin:26.0","env":[{"name":"BITCOIN_ARGS","value":"-regtest=1 -checkmempool=0 -acceptnonstdtxn=1 -debuglogfile=0 -logips=1 -logtimemicros=1 -capturemessages=1 -rpcallowip=0.0.0.0/0 -rpcbind=0.0.0.0 -rest=1 -fallbackfee=0.00001000 -listen=1 -rpcuser=warnet_user -rpcpassword=2themoon -rpcport=18443 -zmqpubrawblock=tcp://0.0.0.0:28332 -zmqpubrawtx=tcp://0.0.0.0:28333 -uacomment=w3"}],"resources":{},"volumeMounts":[{"name":"kube-api-access-7fmrb","readOnly":true,"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount"}],"livenessProbe":{"exec":{"command":["pidof","bitcoind"]},"initialDelaySeconds":5,"timeoutSeconds":1,"periodSeconds":5,"successThreshold":1,"failureThreshold":3},"readinessProbe":{"tcpSocket":{"port":18443},"timeoutSeconds":1,"periodSeconds":1,"successThreshold":1,"failureThreshold":1},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","imagePullPolicy":"IfNotPresent","securityContext":{"capabilities":{"add":["NET_ADMIN","NET_RAW"]},"privileged":true}}],"restartPolicy":"OnFailure","terminationGracePeriodSeconds":30,"dnsPolicy":"ClusterFirst","serviceAccountName":"default","serviceAccount":"default","nodeName":"docker-desktop","securityContext":{},"schedulerName":"default-scheduler","tolerations":[{"key":"node.kubernetes.io/not-ready","operator":"Exists","effect":"NoExecute","tolerationSeconds":300},{"key":"node.kubernetes.io/unreachable","operator":"Exists","effect":"NoExecute","tolerationSeconds":300}],"priority":0,"enableServiceLinks":true,"preemptionPolicy":"PreemptLowerPriority"},"status":{"phase":"Running","conditions":[{"type":"Initialized","status":"True","lastProbeTime":null,"lastTransitionTime":"2024-03-04T15:20:20Z"},{"type":"Ready","status":"True","lastProbeTime":null,"lastTransitionTime":"2024-03-04T15:20:21Z"},{"type":"ContainersReady","status":"True","lastProbeTime":null,"lastTransitionTime":"2024-03-04T15:20:21Z"},{"type":"PodScheduled","status":"True","lastProbeTime":null,"lastTransitionTime":"2024-03-04T15:20:20Z"}],"hostIP":"192.168.65.3","podIP":"10.1.3.54","podIPs":[{"ip":"10.1.3.54"}],"startTime":"2024-03-04T15:20:20Z","containerStatuses":[{"name":"bitcoin","state":{"running":{"startedAt":"2024-03-04T15:20:21Z"}},"lastState":{},"ready":true,"restartCount":0,"image":"bitcoindevproject/bitcoin:26.0","imageID":"docker-pullable://bitcoindevproject/bitcoin@sha256:480272ddc32c6ed5dce3eb9067954939aaee89ddc9799e56cc123fd60e529ac7","containerID":"docker://a8bb0454ec878fdc383fd758c942c7d690b8bfeaf2d8c243f7b4436883cfe82b","started":true}],"qosClass":"BestEffort"}}
2024-03-04T15:20:30+0000 - INFO - 127.0.0.1 - - [04/Mar/2024 15:20:30] "POST /api/0.1 HTTP/1.1" 200 -

**Comment by willcl-ark** [comment]
> Seeing this big response blobs when running tests in K8s, cant figure out why. They look like urllib3 logs but i can see youve set that module to CRITICAL in the config

Seems like I was missing [propagate](https://docs.python.org/3/library/logging.html#logging.Logger.propagate) attribute 🤞🏼

**Comment by pinheadmz** [comment]
On mobile untested ACK. Good find with that propagate flag

**Pull request merged**